core-tests/shared/src/test/scala/zio/ZIOAppSpec.scala

```diff
@@ -55,6 +55,18 @@ object ZIOAppSpec extends ZIOBaseSpec {
         c <- app1.invoke(Chunk.empty).exitCode
         v <- ZIO.succeed(counter.get())
       } yield assertTrue(c == ExitCode.failure) && assertTrue(v == 1)
+    },
+    test("execution of finalizers on interruption") {
+      val finalized = new java.util.concurrent.atomic.AtomicBoolean(false)
+      for {
+        running <- Promise.make[Nothing, Unit]
+        effect   = (running.succeed(()) *> ZIO.never).ensuring(ZIO.succeed(finalized.set(true)))
+        app      = ZIOAppDefault.fromZIO(effect)
+        fiber   <- app.invoke(Chunk.empty).fork
+        _       <- running.await
+        _       <- fiber.interrupt
```

Comment:
`interrupt` gives you the `exit so you can just bind that and there is no need to `join` it on the next line.